browser/brave_rewards/test/rewards_page_browsertest.cc

```diff
@@ -0,0 +1,205 @@
+        &RewardsPageBrowserTest::HandleRequest, base::Unretained(this)));
+  }
+
+  PrefService& GetPrefs() {
+    auto* prefs = browser()->profile()->GetPrefs();
+    CHECK(prefs);
+    return *prefs;
+  }
+
+  RewardsServiceImpl& GetRewardsService() {
+    auto* rewards_service = static_cast<RewardsServiceImpl*>(
+        RewardsServiceFactory::GetForProfile(browser()->profile()));
+    CHECK(rewards_service);
+    return *rewards_service;
+  }
+
+  void LoadScript(const std::string& filename) {
+    std::string script;
+    auto full_path = test_data_dir_.AppendASCII(filename);
+    {
+      base::ScopedAllowBlockingForTesting allow_blocking;
+      ASSERT_TRUE(base::ReadFileToString(full_path, &script));
+    }
+    ASSERT_TRUE(page_contents_);
+    ASSERT_TRUE(content::ExecJs(page_contents_.get(), script));
+  }
+
+  void GivenRewardsIsEnabled() {
+    auto& prefs = GetPrefs();
+    prefs.SetBoolean(prefs::kEnabled, true);
+    prefs.SetString(prefs::kDeclaredGeo, "US");
+    prefs.SetString(prefs::kWalletBrave, R"(
+        {"payment_id":"2b6e71a6-f3c7-5999-9235-11605a60ec93",
+         "recovery_seed":"QgcQHdg6fo53/bGKVwZlL1UkLiql8X7U68jaWgz6FWQ="})");
+  }
+
+  void StartRewardsEngine() {
+    base::RunLoop run_loop;
+    GetRewardsService().StartProcessForTesting(
+        base::BindLambdaForTesting([&]() { run_loop.Quit(); }));
+    run_loop.Run();
+  }
+
+  void RunTests() {
+    ASSERT_TRUE(page_contents_);
+    ASSERT_TRUE(content::ExecJs(page_contents_.get(), "testing.runTests()"));
+  }
+
+  void NavigateToRewardsPage(const std::string& relative_url) {
+    GURL url = GURL(kRewardsPageURL).Resolve(relative_url);
+    ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+    auto* web_contents = browser()->tab_strip_model()->GetActiveWebContents();
+    ASSERT_TRUE(web_contents);
+    page_contents_ = web_contents->GetWeakPtr();
+    LoadScript("test_setup.js");
+  }
+
+  void OpenRewardsPanel() {
+    content::CreateAndLoadWebContentsObserver popup_observer;
+
+    auto* coordinator = RewardsPanelCoordinator::FromBrowser(browser());
+    ASSERT_TRUE(coordinator);
+    ASSERT_TRUE(coordinator->OpenRewardsPanel());
+
+    for (page_contents_.reset(); !page_contents_;) {
+      content::WebContents* web_contents = popup_observer.Wait();
+      GURL url = web_contents->GetLastCommittedURL();
+      if (url.host() == kRewardsPageTopHost) {
+        page_contents_ = web_contents->GetWeakPtr();
+      }
+    }
+
+    LoadScript("test_setup.js");
+  }
+
+ private:
+  std::string GetMethodString(int32_t method) {
+    switch (static_cast<mojom::UrlMethod>(method)) {
+      case mojom::UrlMethod::GET:
+        return "GET";
+      case mojom::UrlMethod::PUT:
+        return "PUT";
+      case mojom::UrlMethod::POST:
+        return "POST";
+      case mojom::UrlMethod::PATCH:
+        return "PATCH";
+      case mojom::UrlMethod::DEL:
+        return "DEL";
+    }
+    return "GET";
+  }
```

Comment:
Looks like it's in `rewards_service_impl.cc`. I'll refactor it out.

Reply:
Rather than create a new util file, I've simply exposed it as a static method of `RewardsServiceImpl` for now. I think the ideal would be to break off the URL-loading parts of `RewardsServiceImpl` into a new class, and this utility method would be homed there.

Reply:
Makes sense, thanks.